packages/contracts-bedrock/contracts/test/OptimismPortal.t.sol

```diff
@@ -526,8 +526,8 @@ contract OptimismPortal_FinalizeWithdrawal_Test is Portal_Initializer {
     function test_finalizeWithdrawalTransaction_differential(
         address _sender,
         address _target,
-        uint64 _value,
-        uint8 _gasLimit,
```

Comment:
The values in the actual transaction type are both uint256, but I get a failure when I set `_gasLimit` to `uint256`. I assume because the `_gasLimit` arg is just too high for the vm?
https://github.com/ethereum-optimism/optimism/blob/ad7fe27b540c173ae7f0ef95e069bebfd007f5b4/packages/contracts-bedrock/contracts/libraries/Types.sol#L53-L54

Reply:
It has to do with the gaslimit check here: https://github.com/ethereum-optimism/optimism/blob/915f3b28791c543961cc34bd16a3732b52637f9e/packages/contracts-bedrock/contracts/L1/OptimismPortal.sol#L191

Reply:
Now that I think about it, I think the solution is to control the amount of gas during this call:

```solidity
op.finalizeWithdrawalTransaction{ value: _tx.value, gas: _tx.gasLimit + FINALIZE_GAS_BUFFER + n }(...)
```
Something like that?

Reply:
That might work... but I assumed that forge would always provide enough gas if it could, so I thought we might be bumping up against some upper bound in what the VM could accept for gasLimit.

Reply:
I think controlling the gas limit passed to the call is more realistic because right now we have no idea what the `gaslimit()` value is returning, I believe it should be considered best practice to pass in a gaslimit to a `CALL` if it uses `gasleft()`

Reply:
Is [this](https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/5105/workflows/df6628a9-bd29-458a-b09e-13d42aa0d652) what you mean? I keep using larger `n` and it's still failing for me.

Reply:
still looking at this, its weird.

Reply:
OK, so after a bunch of testing, I'd like to keep it as is without specifying the gas amount to pass to the call. The reason being that: 
- This amount works: `_tx.gasLimit + 20_000 + 1_000_000`
- This amount fails: `_tx.gasLimit + 20_000 + 500_000`

In the second case, the reason for the failure just seems to be an inaccurate estimation of how much gas actually needs to be provided to the transaction (I verified that there is no difference in actual gas usage between the two cases). 

For this reason, not specifying the gas amount feels much less brittle to me.

Reply:
Ok i'm cool with the implementation that you have. It feels like a todo still to figure out the exact `FINALIZE_GAS_BUFFER`

Reply:
Agree. It's tracked [here](https://linear.app/optimism/issue/ENG-2521/evaluate-and-document-the-purpose-the-relay-gas-buffer-check-in).